docs/guides/single-fetch.md

```diff
@@ -0,0 +1,155 @@
+---
+title: Single Fetch
+---
+
+# Single Fetch
+
+Remix introduced support for "Single Fetch" ([RFC][rfc]) behind the [`future.unstable_singleFetch`][future-flags] flag in [`v2.9.0`][2.9.0] which allows you to opt-into this behavior which will become the default in Remix v3.
+
+## Overview
+
+When you enable Single Fetch, Remix will make a single HTTP call to your server on client-side transitions, instead of multiple HTTP calls in parallel (one per loader). If you are currently returning `Response` instances from your loaders (i.e., `json`/`defer`) then you shouldn't _need_ to make any changes to your app code, but please read through the "breaking" changes below to be aware of some of the underlying behavior changes.
+
+### Breaking Changes
+
+- Single fetch uses a new streaming format under the hood via [`turbo-stream`][turbo-stream], which means that we can stream down more complex data than just JSON, including `Date`, `Map`, `Set`, `Promise`, `RegExp`, and `BigInt` instances
+- Naked objects returned from `loader` and `action` functions are no longer automatically converted into a JSON `Response` and are serialized as-is over the wire
```

Comment:
Is `json` then going to still serialize them? So `return json({ date: new Date() })` and `return { date: new Date() }` are going to return different things?

Reply:
I just read below, yes using `json` will convert the Date to a string

Reply:
So if I understand, if we need to set headers and/or a particular status code to our loader response, we must use `json()` or `defer()` and consequently we can't benefit anymore from what turbo-stream brings (nested promises, Dates, etc). If my assumption is right, maybe this tradeoff users will have to make should be noted somewhere in the docs :)

Reply:
There's a TODO section about headers, from the RFC, the idea was to give a way to add headers inside the loader directly

Reply:
The docs have been updated with the new approach for `status`/`headers`.  The tl;dr; is that `headers()` is deprecated and you will now receive a mutable `response` instance (a "response stub") in your loaders/actions that you set `status`/`headers` on directly - and Remix will combine at the end.  You can still return `Response` instances with single fetch and they should _mostly_ work but you may need to make some small tweaks.  You can then incrementally migrate away from returning `Response` instances to returning naked objects and using the new `response` stub

Reply:
BTW: Is throwing a Response still a thing with single fetch?